### PR TITLE
add size keyword to upsample_nearest

### PR DESCRIPTION
**Pull request by CarloLucibello**
follow up to discussion in https://github.com/FluxML/Flux.jl/pull/1468

### CODE REVIEW

File: src/upsample.jl [hunk 1]
```diff
@@ -3,30 +3,43 @@ export upsample_nearest, ∇upsample_nearest,
...
 ```jldoctest
-julia> upsample_nearest([1 2 3; 4 5 6], (2,3))
+julia> upsample_nearest([1 2 3; 4 5 6], (2, 3))
 4×9 Array{$Int,2}:
  1  1  1  2  2  2  3  3  3
  1  1  1  2  2  2  3  3  3
  4  4  4  5  5  5  6  6  6
  4  4  4  5  5  5  6  6  6
 
+julia> upsample_nearest([1 2 3; 4 5 6], size=(4, 9)); # equivalent to above
```
Comment (mcabbott) [inline src/upsample.jl:25]:
```suggestion
julia> ans == upsample_nearest([1 2 3; 4 5 6]; size=(4, 9))  # equivalent
true
```

File: src/upsample.jl [hunk 2]
```diff
@@ -3,30 +3,43 @@ export upsample_nearest, ∇upsample_nearest,
...
+
 julia> upsample_nearest([1 2 3; 4 5 6], (2,))
 4×3 Array{$Int,1}:
  1  2  3
  1  2  3
  4  5  6
  4  5  6
 ```
+
+julia> upsample_nearest([1 2 3; 4 5 6], size=(4,)); # equivalent to above
```
Comment (mcabbott) [inline src/upsample.jl:35]:
```suggestion
julia> ans == upsample_nearest([1 2 3; 4 5 6], size=(4, 3))
true
```

**Review by mcabbott — APPROVED** [review]
Looks fine. I just suggest making the "same as above" something the computer checks.

File: src/upsample.jl [hunk 3]
```diff
@@ -3,30 +3,45 @@ export upsample_nearest, ∇upsample_nearest,
...
 4×3 Array{$Int,1}:
  1  2  3
  1  2  3
  4  5  6
  4  5  6
 ```
+
+julia> ans == upsample_nearest([1 2 3; 4 5 6], size=(4,))
+true
 """
```
Comment (mcabbott) [inline src/upsample.jl:38]: Not sure I can fix this in the suggestion box, but the fence ``` is out of place here

File: src/upsample.jl [hunk 4]
```diff
@@ -3,30 +3,45 @@ export upsample_nearest, ∇upsample_nearest,
...
+julia> ans == upsample_nearest([1 2 3; 4 5 6]; size=(4, 9))  # equivalent
+true
+
 julia> upsample_nearest([1 2 3; 4 5 6], (2,))
 4×3 Array{$Int,1}:
  1  2  3
  1  2  3
  4  5  6
  4  5  6
 ```
```
Comment (CarloLucibello) [inline src/upsample.jl:34]:
```suggestion
```

File: src/upsample.jl [hunk 5]
```diff
@@ -3,30 +3,44 @@ export upsample_nearest, ∇upsample_nearest,
...
 julia> upsample_nearest([1 2 3; 4 5 6], (2,))
 4×3 Array{$Int,1}:
  1  2  3
  1  2  3
  4  5  6
  4  5  6
-```
+
+julia> ans == upsample_nearest([1 2 3; 4 5 6], size=(4,))
+true
```
Comment (CarloLucibello) [inline src/upsample.jl:36]:
```

Comment (CarloLucibello) [inline src/upsample.jl:36]:
```suggestion
true
```
```

**Pull request merged**